internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriterFactory.java

```diff
@@ -16,94 +14,34 @@
  */
 public class VeniceWriterFactory {
   private final Properties properties;
-  private final String localKafkaBootstrapServers;
-  private final Optional<SharedKafkaProducerService> sharedKafkaProducerService;
+  private final ProducerAdapterFactory producerAdapterFactory;
 
   public VeniceWriterFactory(Properties properties) {
-    this(properties, Optional.empty());
+    this(properties, null, null);
   }
 
-  public VeniceWriterFactory(Properties properties, Optional<SharedKafkaProducerService> sharedKafkaProducerService) {
+  public VeniceWriterFactory(
+      Properties properties,
+      ProducerAdapterFactory producerAdapterFactory,
+      MetricsRepository metricsRepository) {
     this.properties = properties;
-    this.sharedKafkaProducerService = sharedKafkaProducerService;
-    boolean sslToKafka = Boolean.valueOf(properties.getProperty(ConfigKeys.SSL_TO_KAFKA, "false"));
-    if (!sslToKafka) {
-      checkProperty(ConfigKeys.KAFKA_BOOTSTRAP_SERVERS);
-      localKafkaBootstrapServers = properties.getProperty(ConfigKeys.KAFKA_BOOTSTRAP_SERVERS);
-    } else {
-      checkProperty(ConfigKeys.SSL_KAFKA_BOOTSTRAP_SERVERS);
-      localKafkaBootstrapServers = properties.getProperty(ConfigKeys.SSL_KAFKA_BOOTSTRAP_SERVERS);
+    if (metricsRepository != null) {
+      new VeniceWriterStats(metricsRepository);
     }
-  }
-
-  private void checkProperty(String key) {
-    if (!properties.containsKey(key)) {
-      throw new VeniceException(
-          "Invalid properties for Venice writer factory. Required property: " + key + " is missing.");
+    // For now, if VeniceWriterFactory caller does not pass ProducerAdapterFactory, use Kafka factory as default.
+    // Eventually we'll force VeniceWriterFactory creators to inject ProducerAdapterFactory.
+    if (producerAdapterFactory == null) {
+      producerAdapterFactory = new ApacheKafkaProducerAdapterFactory();
     }
+    this.producerAdapterFactory = producerAdapterFactory;
   }
 
   public <K, V, U> VeniceWriter<K, V, U> createVeniceWriter(VeniceWriterOptions options) {
-    // Currently this writerProperties is overloaded as it contains KafkaProducer config and as well as
```

Comment:
Will this break Samza dependency?

Reply:
Although it would definitely be great if we get rid of it once and for all.

Reply:
IMO this should not break samza dep. Not 100% sure though. 

>Although it would definitely be great if we get rid of it once and for all.

We are down to 3 createWrtier methods. I'll raise another PR to remove 2 more and will keep only one of them(`createVeniceWriter(VeniceWriterOptions options)`).